File: regexlib/2021-5-15/python_re2_test_file/regexlib_2782.py
# 2782
# (?i)\w.*\@\w*\.\w*
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"i1@"*5000+"! _1SLQ_2"

import re2 as re
from time import perf_counter

regex = """(?i)\w.*\@\w*\.\w*"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "i1@" * i * 10000 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_2782.js
/* 2782
 * (?i)\w.*\@\w*\.\w*
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"i1@"*5000+"! _1SLQ_2"
 */
var REGEX = /(?i)\w.*\@\w*\.\w*/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'i1@'.repeat(i*10000) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}